src/hooks/globalHooks.js

```diff
@@ -0,0 +1,48 @@
+import { useMediaQuery, useTheme } from '@mui/material';
+import { useRouter } from 'next/router';
+import { useState, useEffect } from 'react';
+
+// Return if embed=true parameter is present in URL
+export function useEmbed() {
+  const router = useRouter();
+  const isEmbed = !!router.asPath.match(/embed=true/g);
+  return isEmbed;
+}
+
+// Get value from localStorage if possible, otherwise return provided default
+function getStorageValue(key, defaultValue) {
+  if (typeof window !== 'undefined') {
+    const value = localStorage.getItem(key);
+    let saved;
+    // log.warn('value:', value);
+    if (value && value !== 'undefined') {
+      saved = JSON.parse(value);
+    }
+    return saved || defaultValue;
+  }
+  return defaultValue;
+}
+
+// Usage: Similar to useState. Accepts a key, value pair. For example:
+// const [mode, setMode] = useLocalStorage('theme', 'light');
+export const useLocalStorage = (key, defaultValue) => {
+  const [value, setValue] = useState(() => getStorageValue(key, defaultValue));
+
+  useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
+  }, [key, value]);
+
+  return [value, setValue];
+};
+
+// Determine if mobile based on theme breakpoints
+export const useMobile = () => {
+  const theme = useTheme();
+  return useMediaQuery(theme.breakpoints.down('sm'));
+};
```

Comment:
If we want to control the mobile status, say, in the customizable web map, we put the whole app into a box and let them render in the mobile size, so in this case, maybe we need somewhere to put the states, and the useMobile will return different value based on the status?

Reply:
@RubenSmn I mean if some day we need to consider do not using iframe, and use component to embed the sandbox instead.

Reply:
Hmm yes true. I think if we'll use an component to render the preview we still need an container and change the height an width of it in order to create a mobile view.
I am not sure is there is a way we could force is by an state.

Reply:
so we would like a way to override the behavior of useMobile? maybe we use the URL, similar to useEmbed, and save that info to context/state only instead of localstorage?

Reply:
Yes something like that

Reply:
im about to work another issue #707 relating to url parameters in general, so i'll definitely include that

Reply:
@okdv I'm not requiring features, I am discussing the solution, so no need to do it if not current necessary, just keep this in mind

Reply:
Now we are using iframe and it works fine